internal/services/controller/controller.go

```diff
@@ -394,3 +413,28 @@ func (c *Controller) debugQueueContent(maxItems int) string {
 
 	return content
 }
+
+func applyInformersIfAvailable(client discovery.DiscoveryInterface, log logrus.FieldLogger, defaultInformers map[reflect.Type]cache.SharedInformer, conditionalInformers ...conditionalInformer) map[reflect.Type]cache.SharedInformer {
+	informersMap := defaultInformers
+	for _, informer := range conditionalInformers {
+		if isResourceAvailable(client, log, informer.groupVersion, informer.apiType) {
+			informersMap[informer.apiType] = informer.informer
+		}
+	}
+	return informersMap
+}
+
+func isResourceAvailable(client discovery.DiscoveryInterface, log logrus.FieldLogger, groupVersion string, kind reflect.Type) bool {
```

Comment:
we won't collect resources until next agent restart, which is like 30mins, this can happen during rebalancing as things like metrics server and other components that register api resources might get kicked around.

If we were not able to get the resource list from the api server, perhaps we should retry in the near future?

Reply:
good suggestion, on it